scripts/migrations/update_domain_codes.py

```diff
@@ -0,0 +1,79 @@
+'''
+update_domain_codes.py
+
+A module that overwrites all of the domain codes with their associated descriptions.
+It also adjusts field lengths if needed.
+'''
+
+from copy import copy
+from json import dumps
+from os.path import dirname, join
+from textwrap import dedent
+
+import arcpy
+
+arcpy.env.workspace = sde = join(dirname(__file__), 'local@electrofishing.sde')
+datasets = arcpy.ListTables() + arcpy.ListFeatureClasses()
+
+domains_in_data = {}
+
+
+print('finding domains and their relationship to data')
+for dataset in datasets:
+    print('\n' + dataset)
+    for field in arcpy.ListFields(dataset):
+        if len(field.domain) == 0 or field.type != 'String':
```

Comment:
should the field type string comparison be upper cased to ignore casing?

Reply:
I don't think so since the types are predefined. Docs: http://pro.arcgis.com/en/pro-app/arcpy/classes/field.htm. It's silly that these types don't match those in AddField.

It wouldn't hurt though. Just don't think that it's necessary.